brigade/core/__init__.py

```diff
@@ -68,6 +70,10 @@ def __init__(self, inventory, dry_run,
             level=logging.ERROR,
             format=format,
         )
+        if available_connections is not None:
+            self.available_connections = available_connections
+        else:
+            self.available_connections = connections.available_connections
```

Comment:
```python
self.available_connection = available_connections or connections.available_connections
```
?

Or perhaps:

```python
self.available_connection  = connections.available_connections
if available_connections:
    self.available_connections.update(available_connections)
```

The default available_connections dict could be named brigade_connections, or core_connections.

Reply:
No, I don't want to force people to use the ones we provide. So if they want to customize it they can do the update themselves. 

I didn't want to do `available_connections or connections.available_connections` in case someone passed `{}` deliberately but I am not sure about the use case so maybe I should do that...

Reply:
But for the second option if someone wanted to overwrite the default `self.available_connections['napalm']` they could just pass in `napalm` as a key in their dict of connections while at the same time preserving the shipped version of the `paramiko` connection.

Reply:
Yeah, but imagine you want to ensure only connections provided by you are used. Because security or compliance or whatever. Easiest would be to just pass a full dict with your modules. Otherwise you'd have to check the ones provided by the `core` and pass them in your own dict with a dummy function or something that triggers an error.

Reply:
I would assume that it would be more common that someone wants to overwrite one of the connection types. If I want to overwrite the paramiko one but keep the napalm one I would need to send in my own paramiko connection but also send in the core napalm connection. For security reasons you can also reach into self.available_connections and delete the other ones.

I think the most common scenario will be that someone has one extra connection type. Like if you had some rest-api which takes a username and password and after that you authenticate with a token. If I brought that to Brigade I might still want to use napalm, paramiko and netmiko, but now I have to spend extra time to recreate them.

I don't think any of these will be super common scenarios though. :)

Reply:
Yeah, but what you are suggesting can be done as:

```
brigade.plugins.tasks.connection.available_connections.update({"napalm": custom_napalm_connection})
```
I don't even need to actually pass it to brigade as it will be read from there by default anyway :P

While the case while I want to remove connections would be with your suggestion:

```
my_connections = {"napalm": my_custom_napalm}
for k in brigade.plugins.tasks.connection.available_connections.keys():
    my_connections[k] = dummy_connection_that_fails

brigade = Brigade(available_connections=my_connections)
```

Reply:
And I agree this is going to be strange (specially disabling connections), if anything the second case and I think it's equally easy to accommodate with both yours and my solution while mine makes less assumptions of how things should work :P